### PR TITLE
fix: process didn't exit successfully

### LINKED ISSUE #10: Process didn't exit successfully
### Description
image

rustc 1.67.1 (d5a82bbd2 2023-02-07)
Microsoft Windows 10.0.19044.2604

### Reproduction
* just clone and cargo run get 

### Try
Try to comment out the following code, then cargo run works
https://github.com/sindresorhus/win-wallpaper/blob/b55c6f6c20d629627d215d43836f9e8eed2547c2/src/lib.rs#L142

I am not good at Rust or C++,  so I try to fix this by the [post](https://devblogs.microsoft.com/oldnewthing/20040520-00/?p=39243).

### PR DESCRIPTION
**Pull request by tangdaoyuan**
Fixes #10

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -139,7 +136,7 @@ impl Drop for DesktopWallpaper {
     fn drop(&mut self) {
         unsafe {
             CoFreeUnusedLibraries();
-            CoUninitialize();
+            // CoUninitialize();
```
Comment (sindresorhus) [inline src/lib.rs:139]: ^

File: src/main.rs [hunk 2]
```diff
@@ -71,5 +72,11 @@ fn main() -> Result<(), String> {
         })?,
     }
 
+    drop(wallpaper); // ensure to release of COM pointers before the CoUninitialize call
+
+    unsafe {
+        CoUninitialize();
+    }
```
Comment (sindresorhus) [inline src/main.rs:79]: This handling should happen in `src/lib.rs`. It's not the responsibility of the CLI part to care about this.

**Comment by sindresorhus** [comment]
@a-isaiahharvey Would you be able to help review this pull request 🙏

**Comment by a-isaiahharvey** [comment]
@sindresorhus I think it's well-organized and follows best practices, including using safe Rust code where possible and making proper use of unsafe Rust code where necessary.

**Comment by sindresorhus** [comment]
@a-isaiahharvey Thanks for the review. Would you be able to send me binaries for https://github.com/sindresorhus/win-wallpaper/commit/07ab304ca23c0f0c2c1f8479b9621cdf38c564c5 ?

**Comment by a-isaiahharvey** [comment]
[wallpaper-aarch64-pc-windows.zip](https://github.com/sindresorhus/win-wallpaper/files/10948845/wallpaper-aarch64-pc-windows.zip)
[wallpaper-x86_64-pc-windows.zip](https://github.com/sindresorhus/win-wallpaper/files/10948846/wallpaper-x86_64-pc-windows.zip)

**Pull request merged**